lnrpc/routerrpc/router_server.go

```diff
@@ -747,3 +752,244 @@ func (s *Server) BuildRoute(ctx context.Context,
 
 	return routeResp, nil
 }
+
+// SubscribeHtlcEvents creates a uni-directional stream from the server to
+// the client which delivers a stream of htlc events.
+func (s *Server) SubscribeHtlcEvents(req *SubscribeHtlcEventsRequest,
+	stream Router_SubscribeHtlcEventsServer) error {
+
+	htlcClient, err := s.cfg.RouterBackend.SubscribeHtlcEvents()
+	if err != nil {
+		return err
+	}
+	defer htlcClient.Cancel()
+
+	for {
+		select {
+		case event := <-htlcClient.Updates():
+			htlcEvent, ok := event.(htlcswitch.HtlcEvent)
+			if !ok {
+				return fmt.Errorf("unexpected event type: %T",
+					event)
+			}
+
+			rpcEvent, err := rpcHtlcEvent(htlcEvent)
+			if err != nil {
+				return err
+			}
+
+			switch e := event.(type) {
```

Comment:
There are now two type switches. Consider making a single marshall function that marshalls all of `htlcEvent`